src/GeneratedHydrator/CodeGenerator/Visitor/HydratorMethodsVisitor.php

```diff
@@ -133,7 +133,7 @@ private function replaceConstructor(ClassMethod $method)
             // Hydrate closures
             $bodyParts[] = "\$this->hydrateCallbacks[] = \\Closure::bind(function (\$object, \$values) {";
             foreach ($propertyNames as $propertyName) {
-                $bodyParts[] = "    if (isset(\$values['" . $propertyName . "'])) {";
+                $bodyParts[] = "    if (isset(\$values['$propertyName']) || \$object->$propertyName !== null && array_key_exists('$propertyName', \$values)) {";
```

Comment:
Please don't use string interpolation - made me super-confused while reading this diff